src/lib/CollisionPrevention/CollisionPrevention.cpp

```diff
@@ -91,8 +91,10 @@ void CollisionPrevention::calculate_constrained_setpoint(Vector2f &setpoint, con
 					float angle = math::radians((float)i * obstacle_distance.increment);
 
 					//max admissible velocity in current bin direction: v = sqrt(2 * a * d), where d is the distance to standstill
-					//a is the constant acceleration and v the current velocity. We use a = a_max/2 to stay well within the limits
-					float vel_max_sqrd = math::max(0.f, max_acc * (distance - _param_mpc_col_prev_d.get()));
+					//a is the constant acceleration and v the current velocity. We use a = a_max/2  and j = j_max/2 to stay well within the limits
+					float acceleration_distance = 2.f * curr_vel * max_acc / _param_mpc_jerk_max.get();
```

Comment:
Passing the norm of the current velocity (`curr_vel = _velocity.norm()` in FlightTaskManualPosition.cpp) may be over-precautious with sideways and rearward obstacles.
I would suggest to pass the current 2D velocity (`Vector2f &curr_vel` instead of `const float curr_vel`), and then project it onto the 2D bin direction.

Reply:
you are right, this makes more sense. No need to limit the velocity that much e.g. if we are already moving away from the obstacle